### PR TITLE
Add `knative/client` and its plugins to the release process

### PR DESCRIPTION
**Pull request by rhuss**
# Changes

🎁  Added instructions for how to release the knative CLI client and selected plugins

-
-
-

/kind <kind>

Fixes #

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,6 +127,17 @@ cd "$(basename "${REPO}" .git)"
 git status
 ```
 
+### Special repository checks
+
+For some repositories some extra manual validation needs to be performed before the release branch is cut:
+
+#### knative/client
+
+* Check that in [version.go](https://github.com/knative/client/blob/main/pkg/kn/commands/version/version.go) the variavle [apiVersions](https://github.com/knative/client/blob/main/pkg/kn/commands/version/version.go#L32) points to the versions of `knative-serving` and `knative-eventing` that is about to be released. You can verify that that the version is picked up properly by calling `./hack/build.sh` from within the client repository.
```
Comment (vaikas) [inline README.md:136]: nit: variavle => variable?

Comment (vaikas) [reply README.md:136]: nit: 'that that the version' => 'that the version"

Comment (vaikas) [reply README.md:136]: Just checking if it's enough to check that particular version (in github version.go) is enough, or does one need to manually checkout the repo and run the command? If the second step is optional, then maybe omit it (or make it optional like below)?

File: README.md [hunk 2]
```diff
@@ -232,6 +244,31 @@ repo in this order:
    (assuming `origin` is the `knative.dev` repo)
 1. Remove the git branch (if any) from the Github UI
 
+### Post-release work
+
+#### Client Homebrew repo
+
+After the client release, the [Homebrew tap](https://github.com/knative/homebrew-client) needs to be updated with the new release:
+
+* Copy `kn.rb` to the `kn@${PREV_RELEASE}.rb` with `$PREV_RELEASE` to be replace with the latest release (e.g. `0.19`).
```
Comment (vaikas) [inline README.md:253]: I wonder if in the future, we can look at productivity to automate these keyed off the release variables there?

File: README.md (hunk 1 above)
Comment (rhuss) [reply README.md:136]: Good point. It should be optional, we have a unit test that checks the proper output.

File: README.md (hunk 2 above)
Comment (rhuss) [reply README.md:253]: Yep, that's on the roadmap https://github.com/knative/homebrew-client/issues/18 

Shouldn't be hard to write at least a script for that, just no time ;-(

**Review by vaikas — APPROVED** [review]
/lgtm
/approve
Let's go! :)

**Pull request merged**